IBAnimatableApp/PresentingViewController.swift

```diff
@@ -0,0 +1,279 @@
+  private let positions = ["Center", "TopCenter", "BottomCenter", "LeftCenter", "RightCenter"]
+  private let sizes = ["Half", "Full"]
+  private let keyboardTranslations = ["None", "MoveUp", "AboveKeyboard"]
+  private let blurEffectStyles = ["None", "ExtraLight", "Light", "Dark"]
+  private let colors = ["0x000000", "0xfe0000", "0xff7900", "0xffb900", "0xffde00", "0xfcff00", "0xd2ff00", "0x05c000", "0x00c0a7", "0x0600ff", "0x6700bf", "0x9500c0", "0xbf0199", "0xffffff"]
+
+  private var selectedButton: UIButton?
+
+  private var selectedAnimationType: String?
+  private var selectedDismissalAnimationType: String?
+  private var selectedModalPosition: String?
+  private var selectedModalWidth: String = "Half"
+  private var selectedModalHeight: String = "Half"
+  private var selectedKeyboardTranslation: String?
+  private var selectedBlurEffectStyle: String?
+
+  // MARK: Life cycle
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+  }
+
+  private func setupModal(presentedViewController: AnimatableModalViewController) {
+    presentedViewController.presentationAnimationType = selectedAnimationType
+    presentedViewController.dismissalAnimationType = selectedDismissalAnimationType
+    presentedViewController.modalPosition = selectedModalPosition ?? "Center"
+    presentedViewController.modalWidth = selectedModalWidth ?? "Half"
+    presentedViewController.modalHeight = selectedModalHeight ?? "Half"
+    presentedViewController.backgroundColor = UIColor(hexString: colors[Int(sliderBackgroundColor.value)])
+    presentedViewController.opacity = CGFloat(sliderOpacity.value)
+    presentedViewController.dismissOnTap = switchDismissOnTap.on
+    presentedViewController.keyboardTranslation = selectedKeyboardTranslation
+    presentedViewController.cornerRadius = CGFloat(sliderCornerRadius.value)
+    presentedViewController.blurEffectStyle = selectedBlurEffectStyle
+    presentedViewController.blurOpacity = CGFloat(sliderBlurOpacity.value)
+    presentedViewController.shadowColor = UIColor(hexString: colors[Int(sliderShadowColor.value)])
+    presentedViewController.shadowOpacity = CGFloat(sliderShadowOpacity.value)
+    presentedViewController.shadowRadius = CGFloat(sliderShadowRadius.value)
+    presentedViewController.shadowOffset = CGPoint(x: CGFloat(sliderShadowOffsetX.value), y: CGFloat(sliderShadowOffsetY.value))
+
+    setDismissalAnimationTypeIfNeeded(presentedViewController)
+  }
+
+  private func setDismissalAnimationTypeIfNeeded(viewController: AnimatableModalViewController) {
+    // FIXME: Dirty hack to make `Flip` and `CrossDissolve` work properly for dismissal transition.
+    // If we don't apply this hack, both the dismissal transitions of `Flip` and `CrossDissolve` will slide down the modal not flip or crossDissolve(fade).
+    if viewController.presentationAnimationType == "Flip" {
```

Comment:
Aaaah, sorry I misunderstood, you are right, we have to do the setup fix only if it's nil